src/Data/MonadicStreamFunction/Util.hs

```diff
@@ -136,6 +136,14 @@ accumulateWith f s0 = feedback s0 $ arr g
   where
     g (a, s) = let s' = f a s in (s', s')
 
+-- | Applies a transfer function to the input and an accumulator,
+-- returning the updated accumulator and output. Equal to
+-- @\f s0 -> feedback s0 $ arr (uncurry f)@.
+mealy :: Monad m => (a -> s -> (b, s)) -> s -> MSF m a b
+mealy f s0 = feedback s0 $ arr g
+  where
+    g (a, s) = let (b, s') = f a s in (b, s')
```

Comment:
Is there any reason not to write `arr $ uncurry f` directly?

Reply:
@turion One would have to examine Core/STG output to give a good judgment.

Reply:
I suggest opting for the simplest, most declarative solution possible by default, and only optimising when there is a real gain.

Reply:
I see, so it might be possible that when writing `arr $ uncurry g$, it will expand `$` and `uncurry` on every tick? Seems plausible...

But what's the benefit of the `let`-construction?

Reply:
@ggreif , Ivan will probably start a research project on measuring and optimising performance of FRP, so maybe we'll gain some insights then.

Reply:
@turion I'm in a similar position: started the research project and I'm evaluating the performance of different FRP constructs.